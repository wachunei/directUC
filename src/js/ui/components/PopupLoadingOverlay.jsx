import styled /* , { keyframes } */ from "styled-components";
import { rgba } from "polished";

// const overlayAnimation = keyframes`
//   from {
//     backdrop-filter: blur(0) grayscale(0);
//   }

//   to {
//     backdrop-filter: blur(1px) grayscale(100%);
//   }
// `;

const Overlay = styled.div`
  z-index: 1;
  position: absolute;
  background: ${(props) => rgba(props.theme.popupBackground, 0.8)};
  margin: 0;
  padding: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  backdrop-filter: blur(1px) grayscale(100%);

  ${"" /* animation: ${overlayAnimation} 0.2s linear forwards; */}
`;

export default Overlay;
