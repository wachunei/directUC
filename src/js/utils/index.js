export const removeSuffix = (suffix, value) =>
  value.endsWith(suffix) ? value.slice(0, -suffix.length) : value;

export const parseDistinguishedName = (name) =>
  name.replace(
    /\w\S*/g,
    (txt) => txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase()
  );

export const isFirefox = !window.chrome.app;
