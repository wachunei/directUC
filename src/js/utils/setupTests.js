// eslint-disable-next-line import/no-extraneous-dependencies
import "jest-styled-components";
