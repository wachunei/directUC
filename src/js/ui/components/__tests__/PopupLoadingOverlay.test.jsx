import React from "react";
import PopupLoadingOverlay from "../PopupLoadingOverlay";
import { renderTree } from "./utils";
import "jest-styled-components";

describe("PopupLoadingOverlay Component", () => {
  test("should match snapshot", () => {
    const tree = renderTree(<PopupLoadingOverlay />);
    expect(tree).toMatchSnapshot();
  });
});
