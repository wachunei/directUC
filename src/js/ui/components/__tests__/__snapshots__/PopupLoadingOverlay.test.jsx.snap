// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopupLoadingOverlay Component should match snapshot 1`] = `
.c0 {
  z-index: 1;
  position: absolute;
  background: rgba(255,255,255,0.8);
  margin: 0;
  padding: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-backdrop-filter: blur(1px) grayscale(100%);
  backdrop-filter: blur(1px) grayscale(100%);
}

<div
  className="c0"
/>
`;
